.github/workflows/main.yml

```diff
@@ -22,34 +22,48 @@ jobs:
       # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
       - uses: actions/checkout@v2
 
-      # Runs a single command using the runners shell
-      - name: Run a one-line script
-        run: echo Hello, world!
-
-      # Runs a set of commands using the runners shell
-      - name: Run a multi-line script
-        run: |
-          echo Add other actions to build,
-          echo test, and deploy your project.
 #############################################################################################
-      - name: Conda setup
+      - uses: conda-incubator/setup-miniconda@v1.7.0
+        with:
+          activate-environment: pulsequantum
+          environment-file: ./environment.yml
+          #python-version: 3.5
+          #condarc-file: etc/example-condarc.yml
+          auto-activate-base: false
+      - name: update_conda
+        shell: bash -l {0}
         run: |
-          conda update -n base conda pip setuptools -y &&
-          conda init --all
-        shell: bash
-      - name: disable_telemetry
-        run: sed -i "s/\(enabled *= *\).*/\1False/" ./pulsequantum/conf/telemetry.ini
-        shell: bash
-      -name: print_bla
-       run: |
-         echo "$(pwd)"
-         echo "$($PATH)"
-         echo "$(conda env list)" 
-      - name: install_package
+          conda update -n base -c defaults conda
+          conda info
+          conda list
+      - name: pip_install_test_requirements 
```

Comment:
Is there any reason for installing conda? It seems like the package is installed with pip anyway and conda is unused

Reply:
conda is installed to setup the environment pulsequantum. 
There is not much in the environment yet, but there will be as the packed evolves. 
Or have I misunderstood?

Reply:
There are two alternative package mangers for python conda and pip. You are first installing conda and then using pip to install the package and the dependencies. That looks a bit odd. I would start with a pure pip powered environment and if I use conda only use conda as much as possible without using pip. 
The plottr envs https://github.com/toolsforexperiments/plottr/blob/master/.github/workflows/python-app.yml and here https://github.com/toolsforexperiments/plottr/blob/master/.github/actions/install-dependencies-and-plottr/action.yml only used pip

Reply:
I see you point :-) I was taking guidance from a azure pipline templat (see below) I will try to mimic the plottr workflow instead . 
```` python
     steps:
    - powershell: | # for conda executable and binaries like sqlite (which pytest-cov uses)
        Write-Host "##vso[task.prependpath]$env:CONDA\Library\mingw-w64\bin"
        Write-Host "##vso[task.prependpath]$env:CONDA\Library\usr\bin"
        Write-Host "##vso[task.prependpath]$env:CONDA\Library\bin"
        Write-Host "##vso[task.prependpath]$env:CONDA\Scripts"
        Write-Host "##vso[task.prependpath]$env:CONDA\bin"
      displayName: Add conda to PATH
    - bash: |
        conda update -n base conda pip setuptools -y &&
        conda init --all
      displayName: Conda setup
    - bash: |
        sed -i "s/\(enabled *= *\).*/\1False/" ./pulsequantum/conf/telemetry.ini
      name: disable_telemetry
      displayName: Disable telemetry
    - bash: |
        conda --version &&
        conda env create --file environment.yml &&
        source activate pulsequantum &&
        pip install -r test_requirements.txt &&
        pip install .
      name: install_package
      displayName: Install Package
````

Reply:
The idea for qcodes was to test with conda on azure pipelines and pip on travis.
However there is only one version of the test and docs requirements since those are not really used by end users